### PR TITLE
Remove languageserver-types dependency in favor of -protocol

### PR DESCRIPTION
**Pull request by colin-grant-work**
#### What it does

The `vscode-languageserver-types` package is part of and reexported by the `vscode-languageserver-protocol` package. There's no good reason to have both, and having both separately could introduce dependency management issues, so this replaces all references to `vscode-languageserver-types` with `vscode-languageserver-protocol`. This will substantially reduce the diff of https://github.com/eclipse-theia/theia/pull/10265.

#### How to test

1. There should be no changes in application behavior - this was a pure search-and-replace operation.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by msujew — COMMENTED** [review]
Looks good. However, I found some references to `vscode-languageserver-types` you seemed to have overlooked:

* [lsp-types.ts](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/src/common/lsp-types.ts#L17)
* [severity.ts](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/src/common/severity.ts#L22)

Also, there is still some reference to it in our [`core/package.json`](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/package.json#L66). I thought the idea was to *not* depend on a specific version of `vscode-languageserver-types`, so I'd remove it there as well.

**Comment by colin-grant-work** [comment]
> Looks good. However, I found some references to `vscode-languageserver-types` you seemed to have overlooked:
> 
> * [lsp-types.ts](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/src/common/lsp-types.ts#L17)
> * [severity.ts](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/src/common/severity.ts#L22)
> 
> Also, there is still some reference to it in our [`core/package.json`](https://github.com/eclipse-theia/theia/blob/678e695297bbb95d73764184d37c935202e35f92/packages/core/package.json#L66). I thought the idea was to _not_ depend on a specific version of `vscode-languageserver-types`, so I'd remove it there as well.

Good catches - thanks. I was so concerned about the shared dependency that that's what I searched for, ignoring `core`.

File: yarn.lock [hunk 1]
```diff
@@ -6681,7 +6681,7 @@ keytar@7.2.0:
   version "7.2.0"
   resolved "https://registry.yarnpkg.com/keytar/-/keytar-7.2.0.tgz#4db2bec4f9700743ffd9eda22eebb658965c8440"
   integrity sha512-ECSaWvoLKI5SI0pGpZQeUV1/lpBYfkaxvoSp3zkiPOz05VavwSfLi8DdEaa9N2ekQZv3Chy+o7aP6n9mairBgw==
-  dependencies:
```
Comment (paul-marechal) [inline yarn.lock:6684]: ?

File: yarn.lock [hunk 2]
```diff
@@ -6681,7 +6681,7 @@ keytar@7.2.0:
...
+    dependencies:
```
Comment (paul-marechal) [inline yarn.lock:6684]: What is this indent doing here?

Comment (colin-grant-work) [reply yarn.lock:6684]: Mostly revealing that I have my git settings set to ignore whitespace changes :-).

**Comment by colin-grant-work** [comment]
@vince-fugnitto, I've added an entry to the changelog, but under 1.22 - no need to rush to get it in before the release on Thursday.

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me :+1:

**Pull request merged**